airflow/executors/base_executor.py

```diff
@@ -171,7 +174,8 @@ def queue_command(
         else:
             self.log.error("could not queue task %s", task_instance.key)
 
-    def queue_workload(self, workload: workloads.All) -> None:
+    @provide_session
```

Comment:
Since you are passing an explicit session in when this is called I would rather we didn't use the `provide_session` decorator (I think it encourages poor session "hygine" as a result)

Reply:
Good point. Applied feedback on other base PR.
Will rebase here once the other is merged.